lang/py/src/avro/io.py

```diff
@@ -909,22 +1045,42 @@ def write_data(self, writers_schema, datum, encoder):
     elif writers_schema.type == 'string':
       encoder.write_utf8(datum)
     elif writers_schema.type == 'int':
-      encoder.write_int(datum)
+      if hasattr(writers_schema, 'logical_type'):
+        if writers_schema.logical_type == constants.DATE:
+          encoder.write_date_int(datum)
+        elif writers_schema.logical_type == constants.TIME_MILLIS:
+          encoder.write_time_millis_int(datum)
+        else:
+          fail_msg = 'Unknown type: %s, logicalType %s' % (writers_schema.type, writers_schema.logical_type)
```

Comment:
I'm not sure if we want to do this; probably ignore the unknown logical types. The spec doesn't specify such case. Could you please check the java implementation to see how they currently handle such case?